rudder-web/src/main/webapp/javascript/rudder/angular/ruleDirectives.js

```diff
@@ -1,7 +1,46 @@
 var ruleDirectives = angular.module('ruleDirectives', []);
 
 ruleDirectives.controller('DirectiveCtrl', ['$scope', function($scope) {
-    $scope.directives = [];
+    $scope.directives = {};
+    
+    //really, angular... 
+    //a call to sortDirectives is needed. 
+    //the array here is ONLY used as a cache so
+    //that angular can know when things don't change.
+    //else, we get infinite looping....
+    $scope.sortedDirectives = [];
+    
+    //needed to correctly display directive by alphaNum
+    $scope.sortDirectives = function() { 
+      var arr = [];
+      for (var o in $scope.directives) {   
+        arr.push({"id":o, "name": $scope.directives[o] });
+      } 
+      var sorted = arr.sort(function(a, b) {
+        return a.name.localeCompare(b.name);
```

Comment:
oh, my bad, this existed already in IE6
http://stackoverflow.com/questions/51165/how-do-you-do-string-comparison-in-javascript